plugins/guests/linux/cap/public_key.rb

```diff
@@ -53,6 +53,7 @@ def self.remove_public_key(machine, contents)
               result=$?
             fi
             rm -f '#{remote_path}'
+            chmod 0600 ~/.ssh/authorized_keys
```

Comment:
I think that it may work if that code is in normal circumstances. but I sometimes use a tool that uses vagrant as a plugin. for example  [test-kitchen](https://github.com/test-kitchen/kitchen-vagrant) and so.At such times, it often happened that an error occurred because the condition could not be satisfied(
I was watching the log, but I have erased it last time and it is not at hand. Sorry).

I think that it is not harm to do this change.